### PR TITLE
[FLINK-28411] - Error Handling for OperatorCoordinatorHolder which may fail Session Cluster

### PR DESCRIPTION
**Pull request by darenwkt**
## What is the purpose of the change

This pull request adds error handling around OperatorCoordinatorHolder.start method, addressing problem raised in [FLINK-28411]. Previously, exception from this method (I.e from starting OperatorCoordinator) bubbles up to Dispatcher to trigger a fatal error which brings down the entire cluster. With this new change, OperatorCoordinatorHolder will instead trigger a globalFailure which is handled by the Scheduler. 

## Brief change log

OperatorCoordinatorHolder.start now handles exception thrown from OperatorCoordinator.start. It rethrows the error if it's a JVM/OOM fatal error. Else, it triggers a globalFailure which is handled by Scheduler to attempt to restart the Job.

## Verifying this change

Added Unit Test to validate:
- Exception is caught and not bubbled up above OperatorCoordinatorHolder.
- GlobalFailure is successfully triggered.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by darenwkt** [comment]
@flinkbot run azure

**Comment by dannycranmer** [comment]
@darenwkt you have a CI failure on compilation step, can you please take a look?

File: mvnvm.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+mvn_version=3.2.5
```
Comment (dannycranmer) [inline mvnvm.properties:1]: What is this for? This is an old maven version, looks like this is not intentional?

File: flink-runtime/src/test/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolderTest.java [hunk 2]
```diff
@@ -442,6 +442,25 @@ public void testCheckpointFailsIfSendingEventFailedBeforeTrigger() throws Except
...
+                            @Override
+                            public void start() {
+                                throw new RuntimeException("Artificial Exception");
+                            }
+                        };
+        final EventReceivingTasks tasks = EventReceivingTasks.createForRunningTasks();
+        final OperatorCoordinatorHolder holder =
+                createCoordinatorHolder(tasks, coordinatorProvider);
+
+        assertNotNull(globalFailure);
```
Comment (dannycranmer) [inline flink-runtime/src/test/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolderTest.java:460]: Can you add an assertion to check that the job is cancelled with the appropriate error?

**Review by dannycranmer — CHANGES_REQUESTED** [review]

**Pull request closed without merging**